### PR TITLE
Support Kafka sasl/scram-256

### PR DESCRIPTION
**Pull request by n313893254**
Proposed changes
======

1. Add 4 fields for backend API added: 
```
 saslPassword
 saslPassword
 saslType
 saslScramMechanism
```
image

2. Refactor Kafka `wiilSave` function.

Types of changes
======

- New feature (non-breaking change which adds functionality)

Linked Issues
======

https://github.com/rancher/rancher/issues/17555

 

Further comments
======

### CODE REVIEW

File: lib/logging/addon/components/logging/target-kafka/component.js [hunk 1]
```diff
@@ -42,39 +51,58 @@ export default Component.extend({
...
 
-    if (brokerType === 'zookeeper') {
-      this.clearCA()
+  // cache and restore
+  brokerTypeChange: observer('brokerType', function() {
+    const t = get(this, 'brokerType');
+    const brokerEndpoints = get(this, 'brokerEndpoints');
+    const zookeeperEndpoint = get(this, 'config.zookeeperEndpoint');
+    const cachedZookeeperEndpoint = get(this, 'cachedZookeeperEndpoint');
+    const cachedBrokerEndpoints = get(this, 'cachedBrokerEndpoints');
```
Comment (westlywright) [inline lib/logging/addon/components/logging/target-kafka/component.js]: brokerEndpoints, cachedZookeeperEndpoint, cachedBrokerEndpoints can all be destructured.

File: lib/logging/addon/components/logging/target-kafka/component.js [hunk 2]
```diff
@@ -42,39 +51,58 @@ export default Component.extend({
...
+
+    if (t === 'zookeeper') {
+      set(this, 'cachedBrokerEndpoints', brokerEndpoints);
+      set(this, 'config.zookeeperEndpoint', cachedZookeeperEndpoint);
+      set(this, 'brokerEndpoints', null);
```
Comment (westlywright) [inline lib/logging/addon/components/logging/target-kafka/component.js]: setProperties

File: lib/logging/addon/components/logging/target-kafka/component.js [hunk 3]
```diff
@@ -42,39 +51,58 @@ export default Component.extend({
...
+    } else if (t === 'broker') {
+      if (!cachedBrokerEndpoints) {
+        this.send('add');
+      } else {
+        set(this, 'brokerEndpoints', cachedBrokerEndpoints);
+      }
+      set(this, 'cachedZookeeperEndpoint', zookeeperEndpoint);
+      set(this, 'config.zookeeperEndpoint', null);
```
Comment (westlywright) [inline lib/logging/addon/components/logging/target-kafka/component.js]: setProperties

**Comment by n313893254** [comment]
@westlywright Done, Thanks for your review.

**Comment by n313893254** [comment]
@westlywright Done, Thanks for your review.

**Pull request merged**